packages/loader/driver-utils/src/readAndParse.ts

```diff
@@ -31,3 +31,29 @@ export function readAndParseFromBlobs<T>(blobs: {[index: string]: string}, id: s
     const decoded = fromBase64ToUtf8(encoded);
     return JSON.parse(decoded) as T;
 }
+
+/**
+ * Utility to retry the read or fetch until it succeeds in it.
+ * @param api - Method to be retried.
+ */
+export async function readWithRetry<T>(api: () => Promise<T>): Promise<T> {
+    let result: T;
+    try {
+        result = await api();
+    } catch (error) {
+        // If it is not retriable, then just throw the error.
+        if (typeof error !== "object" || error.canRetry !== true) {
```

Comment:
There is a function some where that does this check. Can we reuse it?

Reply:
And maybe also we should move handling of IThrottlingWarning into one single API and make sure that all layers (including DeltaManager) use it, such that we if ever need to change format of error objects, we can do it in one place

Reply:
Used the pre existing function. Moved Throttling warning to separate api. Also I believe that throttling warning should be raised on container in all cases. For delta manager we raise it on delta manager, I think we should change that to raise it on container.